pkg/AnewLibrary/Design/Services/DesignService.js

```diff
@@ -198,3 +224,36 @@ const validTarget = (elt, targetId) => {
     }
   }
 };
+
+const updateConnection = (controller, hostId, propName, connection) => {
+  // update connection in live controller
+  const qualifiedConnections = { 
+    [`${designLayerId}$${connection}`]: `${hostId}$${propName}`
+  };
+  Object.assign(controller.connections.inputs, qualifiedConnections);
+  // update connection in graph data
+  const hostSplit = hostId.split('$');
+  const prefixId = hostSplit.slice(2).join('$');
+  const atomName = hostSplit.pop();
+  const host = controller.atoms[hostId];
+  const hostConnections = host.layer.graph[atomName].connections ?? {};
+  const graphQualifiedConnections = {
+    [propName]: `${prefixId}$${connection}`
+  };
+  Object.assign(hostConnections, graphQualifiedConnections);
+  designUpdateTarget(controller, host);
+  Project.ProjectService.SaveProject();
+};
+
+const updateProperty = (controller, hostId, propName, value) => {
+  // update controller state and atoms
+  Controller.writeValue(controller, hostId, propName, value);
```

Comment:
I'll merge and debug.. but just wondering - if the same property has both a value and a connection, what is the expected behaviour?

Reply:
It's always been this way, iiuc. Connection wins.